include/revng/Pypeline/Helpers/Python/AnalysisRunnerImpl.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "nanobind/nanobind.h"
+
+#include "revng/Pypeline/Helpers/AnalysisRunner.h"
+#include "revng/Pypeline/Helpers/Python/RunnerInfo.h"
+#include "revng/Pypeline/Helpers/Python/Utils.h"
+#include "revng/Pypeline/Model.h"
+
+namespace revng::pypeline::helpers::python {
+
+template<typename T>
+struct AnalysisRunnerImpl {
+  static llvm::Error run(T *Handle,
+                         nanobind::handle_t<Model> TheModel,
+                         nanobind::list Containers,
+                         nanobind::list Incoming,
+                         nanobind::str Configuration) {
+    using namespace revng::pypeline::helpers::python;
+
+    Model *CppModel = nanobind::cast<Model *>(TheModel);
+    Request CppIncoming = convertRequests(Incoming);
+    llvm::StringRef CppConfiguration(Configuration.c_str());
+
+    return AnalysisRunner<RunnerInfo<T>>::run(*Handle,
```

Comment:
This should be `AnalysisRunner<T>` and the usage of `RunnerInfo` should be pushed into `AnalysisRunner`.

Reply:
I'm not sure I follow, do you mean to have `AnalysisRunner<T, RunnerInfo>`?

Reply:
```c++
template<typename X>
class AnalysisRunner { ... };
```

Replace all uses of `X` with `RunnerInfo<X>` and use `AnalysisRunner<T>` instead of `AnalysisRunner<RunnerInfo<T>>`.

Reply:
Ah OK, I haven't reviewed the later commit introducing native.